### ISSUE TITLE
Support to add custom cloud init and host class while creating node pool

### ISSUE THREAD

**Issue by sahitgollapudi** [terraform-providers/terraform-provider-oci#860]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

A feature request to add support for providing custom cloud init script and host class (to use hardened images) while creating node pools in OKE.

Basically to add these in the terraform scripts:
 "nodeMetadata": {
  "user_data" : "custom cloud init ",
  "hostclass" : "hostClassName"
  }

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

**Comment by rcohenma** [comment]
Hello,

There is already a node_metadata property in the node_pool resource
https://www.terraform.io/docs/providers/oci/d/containerengine_node_pool.html#node_metadata

Have you attempted to use the user_data key as described here?
https://docs.cloud.oracle.com/iaas/Content/Compute/Concepts/computeoverview.htm

**Comment by bowfeng** [comment]
closed it and feel free to reopen it.
